portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,10 +76,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
       // Get the input from the form
-      String comment = request.getParameter("comment");
-      Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("comment", comment);
-      datastore.put(commentEntity);
+      if (!request.getParameter("comment").isEmpty()) {
+        String comment = request.getParameter("comment");
+        String name;
+        if (!request.getParameter("username").isEmpty()) {
+          name = request.getParameter("username");
+        }
+        else {
+          name = "anonymous";
+        }
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("comment", comment);
+        commentEntity.setProperty("username", name);
+        datastore.put(commentEntity);
+      }
+      else {
+        // Send empty response
+        response.setContentType("");
+        response.getWriter().println();
```

Comment:
I don't think this is actually needed -- it's fine to send a response without content if you're redirecting.